File: 2024/day-1/part_1.py
def main() -> None:
    list_1: list[int] = []
    list_2: list[int] = []

    with open("input", "r", encoding="UTF-8") as file:
        while (line := file.readline()):
            location_1, location_2 = map(int, line.split())

            list_1.append(location_1)
            list_2.append(location_2)
    
    list_1.sort()
    list_2.sort()

    distance: int = 0

    for _, value in enumerate(list_1):
        distance += abs(value - list_2[_])

    print(distance)

    return None


if __name__ == "__main__":
    main()

File: 2024/day-1/part_1.c
#include <stdio.h>
#include <stdlib.h>

int comparator(const void *a, const void *b) {
  return (*(const int *)a > *(const int *)b) -
         (*(const int *)a < *(const int *)b);
}

int main(void) {
  FILE *file = fopen("input", "r");

  if (!file) {
    perror("File error:");

    return EXIT_FAILURE;
  }

  char string[15];
  int list_1[1000], list_2[1000];
  int index = 0;

  while (fgets(string, 20, file)) {
    list_1[index] = atoi(string);
    list_2[index++] = atoi(&string[8]);
  }

  qsort(list_1, 1000, sizeof(int), comparator);
  qsort(list_2, 1000, sizeof(int), comparator);

  int distance = 0;

  for (index = 0; index < 1000; ++index)
    distance += abs(list_1[index] - list_2[index]);

  printf("%d\n", distance);
  fclose(file);

  return EXIT_SUCCESS;
}
